File: 3718. Smallest Missing Multiple of K.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        unordered_set<int> seen;

        for (int num : nums) {
            seen.insert(num);
        }

        int num = k;
        while (seen.count(num)) {
            num += k;
        }

        return num;
    }
};

File: 3718. Smallest Missing Multiple of K.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        HashSet<Integer> seen = new HashSet<>();
        for(int num : nums){
            seen.add(num);
        }
        
        int num = k;
        while(seen.contains(num)){
            num += k;
        }

        return num;
    }
}